src/plugins/discover/public/application/main/utils/fetch_all.ts

```diff
@@ -93,14 +110,23 @@ export function fetchAll(
             recordRawType,
           });
         }
-
-        dataSubjects.documents$.next({
+        const nextDocMsg: DataDocumentsMsg = {
           fetchStatus: FetchStatus.COMPLETE,
           result: records,
           textBasedQueryColumns,
           recordRawType,
           query,
-        });
+          dataView,
+          fetchAppState: appState,
+        };
+        const nextAppState = onResults(nextDocMsg, appStateContainer.getState());
+        if (nextAppState) {
+          nextDocMsg.fetchAppState = nextAppState;
+        }
+        dataSubjects.documents$.next(nextDocMsg);
+        if (nextAppState) {
+          appStateContainer.replaceUrlState(nextAppState);
+        }
```

Comment:
this is the main change, we build the next state required for setting of columns before the send the results to the observable, instead of using a hook afterwards